warehouse/accounts/views.py

```diff
@@ -285,18 +291,49 @@ def request_password_reset(request, _form_class=RequestPasswordResetForm):
 )
 def reset_password(request, _form_class=ResetPasswordForm):
     user_service = request.find_service(IUserService, context=None)
-    user_token_service = request.find_service(IUserTokenService, context=None)
+    token_service = request.find_service(ITokenService, name="password")
 
-    try:
-        token = request.params.get('token')
-        user = user_token_service.get_user_by_token(token)
-    except InvalidPasswordResetToken as e:
-        # Fail if the token is invalid for any reason
-        request.session.flash(e.message, queue="error")
+    def _error(message):
+        request.session.flash(message, queue="error")
         return HTTPSeeOther(
             request.route_path("accounts.request-password-reset"),
         )
 
+    try:
```

Comment:
This block kinda has me wondering if a `UserTokenService` subclassing the `TokenService` makes sense to bundle all this exception handling up into one place. Would this be a better pattern to create in the codebase rather than having somewhat sensitive code that shouldn't change often in the views file, which will?

Reply:
Eh, I'm not sure. This only happens (and will only happen) here in this view, so I'm partial to just keeping it here.

Also, we wouldn't want to be calling `session.flash` or returning redirects from within the service, so we would still need to have some kind of try/except thing happening here anyways, even if it's just catching a generic exception, I think.